modules/subscriptions.php

```diff
@@ -194,15 +197,23 @@ function maybe_send_subscription_email( $new_status, $old_status, $post ) {
 
 		// Only emails posts from these categories
 		if ( ! empty( $only_these_categories ) && ! in_category( $only_these_categories, $post->ID ) ) {
-			update_post_meta( $post->ID, '_jetpack_dont_email_post_to_subs', 1 );
+			$email_to_subs = false;
 		}
 
 		// Email the post, depending on the checkbox option
 		if ( ! empty( $_POST['disable_subscribe_nonce'] ) && wp_verify_nonce( $_POST['disable_subscribe_nonce'], 'disable_subscribe' ) ) {
 			if ( isset( $_POST['_jetpack_dont_email_post_to_subs'] ) ) {
-				update_post_meta( $post->ID, '_jetpack_dont_email_post_to_subs', $_POST['_jetpack_dont_email_post_to_subs'] );
+				$email_to_subs = $_POST['_jetpack_dont_email_post_to_subs'];
```

Comment:
I think this needs to be inverted `$email_to_subs = ! $_POST['_jetpack_dont_email_post_to_subs'];`